source/codegen/common_helpers.py

```diff
@@ -57,17 +59,15 @@ def is_unsupported_enum_array(parameter):
     return not (is_output_parameter(parameter) and is_string_arg(parameter))
   return False
 
-def camel_to_snake(camelString):
+def camel_to_snake(camel_string):
   '''Returns a snake_string for a given camelString.'''
-  camelString = list(camelString)
-  index = 0
-  for x in camelString :
-      if x.isupper():
-          camelString[index] = camelString[index].lower()
-          camelString.insert(index, "_")
-
-      index = index + 1
-  return ("".join(camelString))
+  # Add _ before Words: 
+  # someDeviceIPAddress -> some_DeviceIP_Address
+  s1 = re.sub(r"([^_])([A-Z][a-z]+)", r"\1_\2", camel_string)
```

Comment:
Should this regex also handle numerics in input string (e.g. `voltageLevel2`)

Reply:
Currently: `voltageLevel2` -> `voltage_level2`.
Are you suggesting `voltageLevel2` -> `voltage_level_2`?

I like that change for the numbered enums. But that's also a breaking behavior change, and some of those changes would not be as good. For example: 
`NIDCPOWER_INT32_CURRENT_LIMIT_BEHAVIOR_VAL_CURRENT_TRIP` -> `NIDCPOWER_INT_32_CURRENT_LIMIT_BEHAVIOR_VAL_CURRENT_TRIP`

We could look at that separately potentially.

I think that this method is susceptible to the same issue you caught below:

`establishHTTP2Connection` -> `establish_htt_p2_connection` -> BAD
`establishHTTP2Connection` -> `establish_http2_connection` -> GOOD/PROPOSED
`establishHTTP2Connection` -> `establish_http_2_connection` -> Also OK

Reply:
Ah, I didn't think so far ahead. I couldn't figure out where numbers are being handled in you regex, so my comment was mostly regarding that. But looks like it does take care of numbers as well and create your PROPOSED string above. That handling looks good to me compared to adding `_` before the number as well.